src/silx/io/test/test_url.py

```diff
@@ -27,190 +27,236 @@
+
+
+def test_slice():
+    url = DataUrl("/a.h5?path=/b&slice=5,1")
+    expected = [True, True, None, "/a.h5", "/b", (5, 1)]
+    assertUrl(url, expected)
+
+
+def test_slice2():
+    url = DataUrl("/a.h5?path=/b&slice=2:5")
+    expected = [True, True, None, "/a.h5", "/b", (slice(2, 5),)]
+    assertUrl(url, expected)
+
+
+def test_slice3():
+    url = DataUrl("/a.h5?path=/b&slice=::2")
+    expected = [True, True, None, "/a.h5", "/b", (slice(None, None, 2),)]
+    assertUrl(url, expected)
+
+
+def test_slice_ellipsis():
+    url = DataUrl("/a.h5?path=/b&slice=...")
+    expected = [True, True, None, "/a.h5", "/b", (Ellipsis, )]
+    assertUrl(url, expected)
+
+
+def test_slice_slicing():
+    url = DataUrl("/a.h5?path=/b&slice=:")
+    expected = [True, True, None, "/a.h5", "/b", (slice(None), )]
+    assertUrl(url, expected)
+
+
+def test_slice_missing_element():
+    url = DataUrl("/a.h5?path=/b&slice=5,,1")
+    expected = [False, True, None, "/a.h5", "/b", None]
+    assertUrl(url, expected)
+
+
+def test_slice_no_elements():
+    url = DataUrl("/a.h5?path=/b&slice=")
+    expected = [False, True, None, "/a.h5", "/b", None]
+    assertUrl(url, expected)
+
+
+def test_create_relative_url():
+    url = DataUrl(scheme="silx", file_path="./foo.h5", data_path="/", data_slice=(5, 1))
+    assert not url.is_absolute()
+    url2 = DataUrl(url.path())
+    assert url == url2
+
+
+def test_create_absolute_url():
+    url = DataUrl(scheme="silx", file_path="/foo.h5", data_path="/", data_slice=(5, 1))
+    url2 = DataUrl(url.path())
+    assert url == url2
+
+
+def test_create_absolute_windows_url():
+    url = DataUrl(scheme="silx", file_path="C:/foo.h5", data_path="/", data_slice=(5, 1))
+    url2 = DataUrl(url.path())
+    assert url == url2
+
+
+def test_create_slice_url():
+    url = DataUrl(scheme="silx", file_path="/foo.h5", data_path="/", data_slice=(5, 1, Ellipsis, slice(None)))
+    url2 = DataUrl(url.path())
+    assert url == url2
+
+
+def test_wrong_url():
+    url = DataUrl(scheme="silx", file_path="/foo.h5", data_slice=(5, 1))
+    assert not url.is_valid()
+
+
+@pytest.mark.parametrize("data",
+    [
+        (1, "silx:///foo.h5?slice=1"),
+        ((1,), "silx:///foo.h5?slice=1"),
+        (slice(None), "silx:///foo.h5?slice=:"),
+        (slice(1, None), "silx:///foo.h5?slice=1:"),
+        (slice(None, -2), "silx:///foo.h5?slice=:-2"),
+        (slice(1, None, 3), "silx:///foo.h5?slice=1::3"),
+        (slice(None, 2, 3), "silx:///foo.h5?slice=:2:3"),
+        (slice(None, None, 3), "silx:///foo.h5?slice=::3"),
+        (slice(1, 2, 3), "silx:///foo.h5?slice=1:2:3"),
+        ((1,slice(1,2)), "silx:///foo.h5?slice=1,1:2"),
+    ]
+)
+def test_path_creation(data):
+    """make sure the construction of path succeed and that we can
+    recreate a DataUrl from a path"""
+    data_slice, expected_path = data
+    url = DataUrl(scheme="silx", file_path="/foo.h5", data_slice=data_slice)
+    path = url.path()
+    DataUrl(path=path)
```

Comment:
This is not used, is it a typo?

Reply:
It is used, it checked that the url can be reparsed.

It's part of the previous implementation